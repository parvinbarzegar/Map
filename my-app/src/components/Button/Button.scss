.button {
    font-size: 15px;
    line-height: 15px;
    font-family: "Helvetica Neue",Helvetica,Arial,sans-serif;
    margin: 0;
    cursor: pointer;
    display: inline-block;
    font-weight: 700;
    line-height: 12px;
    position: relative;
    text-align: center;
    transition: all .15s linear;
    background-color: #ff7a59;
    border-color: #ff7a59;
    color: #fff;
    border-radius: 3px;
    border-style: solid;
    border-width: 1px;
    font-size: 14px;
    padding: 12px 24px;
    width: 120px;
    &--primary{
        background: #00a662;
    border-color: #00a662;
    color: #ffffff;
    }
    &--secondry{
        color: red;
    }
    &--loading{
        color:yellow;
    }
}