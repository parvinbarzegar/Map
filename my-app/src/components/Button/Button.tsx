import React from "react";
import cls from "classnames";
import "./Button.scss";

interface IButtonProp {
  children: string;
  onClick?: Function;
  type?: "primary" | "secondry" | "loading";
}

const Button: React.FunctionComponent<IButtonProp> = ({
  children,
  onClick,
  type = "primary",
}) => (
  <button
    onClick={()=>onClick?.()}
    className={cls(
      "button",
      { "button--primary": type === "primary" },
      { "button--secondry": type === "secondry" },
      { "button--loading": type === "loading" }
    )}
  >
    {children}
  </button>
);
export default Button;
