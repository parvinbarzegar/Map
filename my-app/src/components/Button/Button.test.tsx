import React from 'react';
import { screen , render, cleanup } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Button from './Button';

describe('Test suits for <Button/>',()=> {
    beforeEach(()=> {
        cleanup();
    })
    it('should match to snapshot',()=> {
        const { container } = render(<Button>Hope</Button>);
        expect(container).toMatchSnapshot();
    })
    it('should render a text',()=> {
        render(<Button>Hello</Button>);
        const btn = screen.getByRole('button');
        expect(btn).toHaveTextContent('Hello');
    })
    it('should call function when it is clicked',async()=> {
        const myClick = jest.fn();
        render(<Button onClick={()=>myClick}>hello</Button>);
        const btn = screen.getByRole('button');
        await userEvent.click(btn);
        expect(myClick).toHaveBeenCalled();
    })
    it('should be a primary button',()=> {
        render(<Button type="primary">hello</Button>)
        const btn = screen.getByRole('button');
        expect(btn).toHaveClass('button');
        expect(btn).toHaveClass('button--primary');
    })
    it('should be a secondry button',()=> {
        render(<Button type="secondry">hello</Button>)
        const btn = screen.getByRole('button');
        expect(btn).toHaveClass('button');
        expect(btn).toHaveClass('button--secondry');
    })
    it('should be a loading button',()=> {
        render(<Button type="loading">hello</Button>)
        const btn = screen.getByRole('button');
        expect(btn).toHaveClass('button');
        expect(btn).toHaveClass('button--loading');
    })

})