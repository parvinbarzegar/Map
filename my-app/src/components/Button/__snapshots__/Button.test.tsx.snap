// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suits for <Button/> should match to snapshot 1`] = `
<div>
  <button
    class="button button--primary"
  >
    Hope
  </button>
</div>
`;
